h/static/scripts/forms-common/components/Form.tsx

```diff
@@ -0,0 +1,33 @@
+import type { ComponentChildren } from 'preact';
+import type { JSX } from 'preact';
+
+export type FormProps = JSX.FormHTMLAttributes & {
+  /**
+   * CSRF token to render in a hidden field.
+   *
+   * If the form is not going to be submitted to the backend directly, for
+   * example because the frontend is going to make API requests instead, this
+   * can be set to `null`. This prop is required so that callers have to make
+   * an explicit choice.
```

Comment:
Maybe instead of making this prop required with value `null`, we could have two form components, one with the token and one without the token, and give them clear names.

That way there's still a need to make an explicit decision, and the meaning is more obvious than `csrfToken={null}`, because of the name of the form component.

Maybe something like `ClientForm` and `ServerForm` could do. We can make `ServerForm` wrap `ClientForm` with the addition of the token input to avoid duplication.